pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1220,7 +1221,7 @@ private void asyncDeleteCursorWithClearDelayedMessage(String subscriptionName,
                     if (ex != null) {
                         unsubscribeFuture.completeExceptionally(ex);
                     } else {
```

Comment:
Should we delete the cursor after the in-progress compaction task is finished?

Reply:
I think we should stop the compaction if the topic is deleted.
After checking the code, it looks like the RawReader will create the topic again if the topic auto-creation is enabled on the broker-side. Will it cause a topic deletion failure if there is a running compactor?

Reply:
@poorbarcode @codelipenghui 

> Should we delete the cursor after the in-progress compaction task is finished?

If there is a compaction task in progress and RawRead is already connected to the topic, then the topic will not be deleted.

> it looks like the RawReader will create the topic again if the topic auto-creation is enabled on the broker-side. Will it cause a topic deletion failure if there is a running compactor

I changed the `createTopicIfDoesNotExist` of RawReader to false and added a check if the topic is closing or deleting before triggering compaction. So, the case of recreating the topic should not happen.

Reply:
> If there is a compaction task in progress and RawRead is already connected to the topic, then the topic will not be deleted.

Force delete topic will ignore the connected consumers.

Reply:
> Force delete topic will ignore the connected consumers.

Force delete topic will disconnect all connections then the compaction task will fail. I think we can ignore this case.

Reply:
@coderzc 

> Force delete topic will disconnect all connections, and the compaction task will fail. I think we can ignore this case.

Maybe the in-progress Compaction task is trying to update the `cursor.props`?  For example:
- The Last ledger of Compacted data is `1`
- Start Compaction task
- Delete topic
  - delete ledger `1`
- Update the last ledger of Compacted data to `2`
  - update success
- Delete cursor success

There is an orphan ledger `2`

Reply:
> Should we delete the cursor after the in-progress compaction task is finished?

I agree. We should first stop the compaction, then delete the cursors.

And it's better to add a test for topic force deletion.

Reply:
@poorbarcode I added wait for current compaction finished before delete the cursor, and due to `cursor.props` will be updated after opening compaction ledger, so I get compactionLedgerId from `compactedTopicContext` and set `ackReceiptEnabled=true` of RawReader.

Reply:
> Could you add a test that covers the case that @poorbarcode mentioned?

Yes, I will add a test